src/Reflection/ReflectionClass.php

```diff
@@ -236,7 +238,16 @@ public static function createFromNode(
      */
     public function getShortName() : string
     {
-        return $this->node->name;
+        if (!$this->isAnonymous()) {
+            return $this->node->name;
+        }
+
+        $fileName = $this->getFileName();
+        if ($fileName === null) {
```

Comment:
This could cause collisions when evaluating multiple classes, no?

Even in code with no assigned file:

```php
<?php

eval('$a = new class() {};');
eval('$b = new class() {};');

var_dump(get_class($a), get_class($b));
```
```
string(58) "class@anonymous/in/ZB2DN(3) : eval()'d code0x7fb03707a07a"
string(58) "class@anonymous/in/ZB2DN(4) : eval()'d code0x7fb03707a07a"
```
https://3v4l.org/ZB2DN